pages/clubs/Running.jsx

```diff
@@ -43,40 +43,40 @@ var Running = React.createClass({
   componentDidMount: function() {
     this.fetchJSON("https://mozilla.github.io/learning-networks/clubs/clubs-resources.json",this.setData);
   },
-  getLinks : function(){
-    var categories = this.state.guides.map(function(category){
-      var guideLinks = category.guides.map(function(guide){
-
-        if(guide.translations && guide.translations.length > 0){
-          var translationLinks = guide.translations.map(function(translation){
-            return (
-              <li>
-                <a href={ translation.url }>{ translation.title }</a>
-                <span className="language"> ({ translation.language })</span>
-              </li>
-            );
-          });
-        }
-
+  getLinkForGuide: function(guide) {
+    if(guide.translations && guide.translations.length > 0){
+      var translationLinks = guide.translations.map(function(translation){
         return (
-          <li>
-            <a href={guide.url}>{ guide.title }</a>
-            { translationLinks ? <ul className="translations">{ translationLinks }</ul> : "" }
+          <li key={translation.title}>
+            <a href={ translation.url }>{ translation.title }</a>
+            <span className="language"> ({ translation.language })</span>
           </li>
         );
       });
+    }
 
-      return (
-        <section className="resourceCategory">
-          <h3> { category.category } </h3>
-          <ul>
-            { guideLinks }
-          </ul>
-        </section>
-      );
-    });
+    return (
+      <li key={guide.title}>
+        <a href={guide.url}>{ guide.title }</a>
+        { translationLinks ? <ul className="translations">{ translationLinks }</ul> : "" }
+      </li>
+    );
+  },
+  getGuideLinks: function(category) {
+    var guideLinks = category.guides.map(this.getLinkForGuide);
+    var cat = category.category;
 
-    return categories;
+    return (
+      <section className="resourceCategory" key={cat}>
+        <h3> { cat } </h3>
+        <ul>
+          { guideLinks }
+        </ul>
+      </section>
+    );
+  },
+  getGuideList: function() {
+    return this.state.guides.map(this.getGuideLinks);
```

Comment:
two reasons really, it was a massive map-map-map block and poorly named, and by doing it this way the `this` context is preserved due to React policy rather than possibly getting switch to something else. I'd rather have it a little over abstracted than three nested maps.